src/behavior-considered-undefined.md

```diff
@@ -60,6 +60,10 @@ code.
     > `rustc_layout_scalar_valid_range_*` attributes.
 * Incorrect use of inline assembly. For more details, refer to the [rules] to
   follow when writing code that uses inline assembly.
+* **In `const` context** (i.e., during the evaluation of a `const`/`static`
+  initializer, array length, enum discriminant, or const generic value):
+  transmuting or otherwise reinterpreting a pointer into some allocated object
+  as an integer or other non-pointer type.
```

Comment:
Pointer type should ~~probably~~ maybe link to https://doc.rust-lang.org/reference/types/pointer.html

I say maybe because are those three types (`&T`, `&mut T`, `*T`) the only pointer types, or are function pointers also included? If so, we should probably reorganize the types chapter so that all pointer types as per this definition are in the `types/pointer.html` chapter.

Reply:
For the purpose of CTFE, function pointers are definitely pointers.

But for the semantics in general this is less clear so I would hold off on changing the overall organization.